gangplank/ocp/worker.go

```diff
@@ -240,7 +243,18 @@ func (ws *workSpec) Exec(ctx ClusterContext) error {
 		next, _, _ := cosa.ReadBuild(cosaSrvDir, "", cosa.BuilderArch())
 		if next != nil && next.BuildArtifacts != nil && (mBuild.BuildArtifacts == nil || mBuild.BuildArtifacts.Ostree.Sha256 != next.BuildArtifacts.Ostree.Sha256) {
 			log.Debug("Stage produced a new OStree")
-			if err := uploadCustomBuildContainer(ctx, ws.APIBuild, next.BuildID); err != nil {
+
+			// push for other defined registries
+			for _, v := range ws.JobSpec.PublishOscontainer.Registries {
+				if err := pushOstreeToRegistry(ctx, &v, next); err != nil {
+					log.WithError(err).Warningf("Push to registry %s failed", v.URL)
+					return err
+				}
+			}
+
+			// push for custom build strategy
+			if err := uploadCustomBuildContainer(ctx, ws.JobSpec.PublishOscontainer.BuildStrategyTLSVerify, ws.APIBuild, next); err != nil {
```

Comment:
If it's running as a build strategy, it will respect the API push specification in the build config. This is targeted at Prow/E2E testing. Since most of our runs won't be done using a Custom Build Strategy is a noop. If you're running it as a custom build strategy then its assumed that you intended to push.